Refactor SFCC SDK implementation and configuration

darekrossman opened the pull request:
- swapped out `commerce-sdk` with `commerce-sdk-isomorphic`
- cleaned up some type defs, using sdk types
- removed `scapi.ts`
- removed `ocapi.ts`
- reworked `getProductRecommendations` to use a scapi category search 

NOTE: I noticed the project has no prettier config, and formatting has become inconsistent across several files. Per discussion with Alex, I'm only applying **default** prettier formatting to files I modify for now. This will be worked out later.

darekrossman committed:
Refactor SFCC SDK implementation and configuration

darekrossman committed:
Refactor product recommendation logic and remove unused OCAPI module

ah100101 commented on lib/sfcc/index.ts:
```diff
@@ -1,36 +1,30 @@
-    clientId: process.env.SFCC_CLIENT_ID,
-    organizationId: process.env.SFCC_ORGANIZATIONID,
-    shortCode: process.env.SFCC_SHORTCODE,
-    siteId: process.env.SFCC_SITEID,
+    clientId: process.env.SFCC_CLIENT_ID || "",
+    organizationId: process.env.SFCC_ORGANIZATIONID || "",
+    shortCode: process.env.SFCC_SHORTCODE || "",
+    siteId: process.env.SFCC_SITEID || "",
   },
```
Hey @darekrossman - To help guide a team setting this up can we throw an error if any of these environment variables are undefined?

darekrossman replied:
Sure. I added a check that would throw an error at this point - but it wouldnt actually throw until a request is made. Ive pushed a different approach by adding an `instrumentation.ts` file to stop the process before it starts. Let me know your thoughts on this - I can switch it back to a runtime check. 

<img width="620" alt="Screenshot 2025-03-24 at 2 23 27 PM" src="https://github.com/user-attachments/assets/0a7f0e85-f07f-43e3-a55a-62f09cc5565f" />

ah100101 commented on lib/sfcc/index.ts:
```diff
@@ -268,36 +258,18 @@ export async function updateCart(
-  const sortedResults = recommendedProducts
-    .sort((a: any, b: any) => a.index - b.index)
-    .map((item) => item.productResult);
+  const products = await searchProducts({ categoryId, limit: 11 });
```
Hey @darekrossman - I like this searchProducts as the recommendation approach, can we wrap this in unstable_cache to help with performance?

darekrossman replied:
Ive retooled `getCollectionProducts` and used that for simple category search with unstable cache

ah100101 commented on lib/sfcc/index.ts:
```diff
@@ -547,17 +515,19 @@ function reshapeProduct(product: SalesforceProduct.ShopperProducts.Product) {
+          return {
+            id: attribute.id,
+            name: attribute.name!,
+            // TODO: might be a better way to do this, we are providing the name as the value
```
I added this comment originally due to how the frontend is implemented, I think we can remove this now.

ah100101 reviewed (changes requested):
Great PR! I left a couple minor comments, let me know if you have any questions or concerns!

darekrossman commented:
✅ I've addressed this feedback by:
1. Ensuring the `validateEnvironmentVariables` function throws clear errors when any required environment variables are missing
2. Added proper TypeScript types with `SFCCSDKParameters` interface
3. Returning properly formatted parameters for the SDK

The error message now includes a link to the documentation and lists all missing variables, which should help teams during setup.

darekrossman committed:
Add environment variable registration function to instrumentation

darekrossman committed:
node runtime check

darekrossman committed:
update and use getCollectionProducts for cached category search

darekrossman committed:
Remove TODO comment regarding product attribute name handling in reshapeProduct function

darekrossman committed:
Merge branch 'main' into sdk-replacement and some cleaning

darekrossman committed:
Remove validateEnvironmentVariables function from utils.ts

darekrossman committed:
fix ts error
